src/shared/sysctl-util.c

```diff
@@ -9,28 +9,37 @@
 #include "fileio.h"
 #include "log.h"
 #include "macro.h"
+#include "path-util.h"
 #include "string-util.h"
 #include "sysctl-util.h"
 
 char *sysctl_normalize(char *s) {
         char *n;
 
         n = strpbrk(s, "/.");
+
         /* If the first separator is a slash, the path is
          * assumed to be normalized and slashes remain slashes
          * and dots remains dots. */
-        if (!n || *n == '/')
-                return s;
-
-        /* Otherwise, dots become slashes and slashes become
-         * dots. Fun. */
-        while (n) {
-                if (*n == '.')
-                        *n = '/';
-                else
-                        *n = '.';
-
-                n = strpbrk(n + 1, "/.");
+
+        if (n && *n == '.')
+                /* Dots become slashes and slashes become dots. Fun. */
+                while (n) {
```

Comment:
doesn't matter, but we already know `n` is set on the first iteration, hence could become a `do {} while`  loop